app/lib/server/server.go

```diff
@@ -0,0 +1,474 @@
+package server
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/user"
+	"path"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dedis/cothority/app/lib/config"
+	"github.com/dedis/cothority/lib/crypto"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+
+	"regexp"
+
+	// Empty imports to have the init-functions called which should
+	// register the protocol
+	_ "github.com/dedis/cosi/protocol"
+	_ "github.com/dedis/cosi/service"
+	"github.com/dedis/cothority/app/lib/ui"
+	"github.com/dedis/cothority/lib/cosi"
+	s "github.com/dedis/cothority/services/cosi"
+	"github.com/dedis/crypto/abstract"
+	crypconf "github.com/dedis/crypto/config"
+)
+
+// DefaultServerConfig is the name of the default file to lookup for server
+// configuration file
+const DefaultServerConfig = "config.toml"
+
+// DefaultGroupFile is the name of the default file to lookup for group
+// definition
+const DefaultGroupFile = "group.toml"
+
+// DefaultPort where to listen; At time of writing, this port is not listed in
+// /etc/services
+const DefaultPort = 6879
+
+// DefaultAddress where to be contacted by other servers
+const DefaultAddress = "127.0.0.1"
+
+// Service used to get the port connection service
+const whatsMyIP = "http://www.whatsmyip.org/"
+
+// How long we're willing to wait for a signature
+var RequestTimeOut = time.Second * 1
+
+// interactiveConfig will ask through the command line to create a Private / Public
+// key, what is the listening address
+func InteractiveConfig(binaryName string, ed25519 bool) {
```

Comment:
Instead of a `bool`, I would have prefered a separate function that clearly indicates the difference, since it is *hack*. But we still have to discuss with Bryan about these ed25519 issues so it may not be necessary after (if for example we implement a *real* Ed25519 suite)